src/providers/providerProvider.ts

```diff
@@ -0,0 +1,126 @@
+import * as vscode from 'vscode';
+import { Utils } from 'vscode-uri';
+import { ExecuteCommandParams, ExecuteCommandRequest } from 'vscode-languageclient';
+
+import { ClientHandler } from '../clientHandler';
+import { getActiveTextEditor } from '../vscodeUtils';
+
+interface ModuleProvidersResponse {
+  v: number;
+  provider_requirements: {
+    [provider: string]: {
+      display_name: string;
+      version_constraint?: string;
+      docs_link?: string;
+    };
+  };
+  installed_providers: {
+    [provider: string]: string;
+  };
+}
+
+class ModuleProvider extends vscode.TreeItem {
```

Comment:
I think we discussed this in a different context and didn't come to a decision, but naming the treeitem `ModuleProvider` here confuses me with the `ModuleProvider` treedataprovider in `moduleProvider.ts`.

Maybe we need to append each thing with `TreeItem` and `TreeDataProvider` to disambiguate these classes since we overlap with Terraform terminology.

Reply:
You're right; the naming is confusing. I've updated the class names and renamed the files. Do the new names work for you?